### PR TITLE
Add NVMe-oF support

### PR DESCRIPTION
**Pull request by qwren**
This patch will add the support for NVMe-oF.

Signed-off-by: Shixin Yang <shixin.yang@intel.com>
Signed-off-by: Qiaowei Ren <qiaowei.ren@intel.com>

### CODE REVIEW

File: csi/server/plugin/opensds/controller.go [hunk 1]
```diff
@@ -1251,6 +1260,17 @@ func extractISCSIInitiatorFromNodeInfo(nodeInfo string) (string, error) {
...
 
+func extractNvmeofInitiatorFromNodeInfo(nodeInfo string) (string, error) {
+	for _, v := range strings.Split(nodeInfo, ",") {
+		if strings.Contains(v, "nqn") {
+			glog.Info("Nvmeof initiator is ", v)
+			return v, nil
+		}
+	}
+
+	return "", errors.New("No Nvmeof initiators found")
```
Comment (leonwanghui) [inline csi/server/plugin/opensds/controller.go]: Please use lower case letter for error message.

**Comment by xing-yang** [comment]
Please resolve CI failures.

**Comment by qwren** [comment]
> Please resolve CI failures.

According to Leon, this will depend on Hotpot release ant this CI should pass after that.

**Comment by xing-yang** [comment]
@qwren can you add a unit test?

**Comment by xing-yang** [comment]
Hi @qwren can you please update this PR?  Hotpot m-3 has been released which includes the NVMeoF connector change.  Thanks.

**Pull request closed without merging**